modules/data-entry/src/main/frontend/src/questionnaireEditor/AnswerOptions.jsx

```diff
@@ -28,77 +28,68 @@ import {
 } from "@material-ui/core";
 
 import QuestionnaireStyle from '../questionnaire/QuestionnaireStyle';
-import { v4 as uuidv4 } from 'uuid';
 import CloseIcon from '@material-ui/icons/Close';
+import { stringToHash } from "../escape.jsx";
 
 let AnswerOptions = (props) => {
-  const { data, path, saveButtonRef } = props;
+  const { data, path, saveButtonRef, classes } = props;
   let [ options, setOptions ] = useState(Object.values(data).filter(value => value['jcr:primaryType'] == 'lfs:AnswerOption').slice());
-  let [ newUUID, setNewUUID ] = useState([]);
-  let [ newValue, setNewValue ] = useState([]);
   let [ deletedOptions, setDeletedOptions ] = useState([]);
-  let [ tempValue, setTempValue ] = useState(''); // Holds new, non-comitted answer options
+  let [ tempValue, setTempValue ] = useState(''); // Holds new, non-committed answer options
+  let [ isDuplicate, setIsDuplicate ] = useState(false);
 
   // Clear local state when data changes
   useEffect(() => {
     setOptions(Object.values(data).filter(value => value['jcr:primaryType'] == 'lfs:AnswerOption').slice());
-    setNewUUID([]);
-    setNewValue([]);
     setDeletedOptions([]);
     setTempValue('');
+    setIsDuplicate(false);
   }, [data])
 
-  let deleteOption = (value) => {
-    setOptions(oldOptions => {
-      let newOptions = oldOptions.slice();
-      newOptions.splice(newOptions.indexOf(value), 1);
-      return newOptions;
-    })
-
+  let deleteOption = (index) => {
     setDeletedOptions(old => {
       let newDeletedOptions = old.slice();
-      newDeletedOptions.push(value);
+      newDeletedOptions.push(options[index]);
       return newDeletedOptions;
-    })
-  }
+    });
 
-  let updateInsertedOption = (index, event) => {
-    setNewValue(oldValue => {
-      var value = oldValue.slice();
-      value[index] = event.target.value;
-      return value;
-    })
+    setOptions(oldOptions => {
+      let newOptions = oldOptions.slice();
+      newOptions.splice(index, 1);
+      return newOptions;
+    });
   }
 
-  let deleteInsertedOption = (index, event) => {
-    setNewValue(oldValue => {
-      let value = oldValue.slice();
-      value.splice(index, 1);
-      return value;
-    })
-    setNewUUID(oldUuid => {
-      let uuid = oldUuid.slice();
-      uuid.splice(index, 1);
-      return uuid;
-    });
+  let validateOption = (optionInput) => {
+    if (optionInput) {
+      setIsDuplicate(false);
+      let inputs = (optionInput + ' ').split(/=(.+)/);
+      let duplicateOption = options.find( option => option.value.trim() === inputs[0].trim() || inputs[1] && (option.label?.trim() === inputs[1].trim()));
```

Comment:
With the change above most of the `trim()`s on this line can probably be removed. To be confirmed.
```suggestion
      let duplicateOption = options.find( option => option.value === inputs[0] || inputs[1] && (option.label === inputs[1]));
```